### DISCUSSION TITLE
Expansion of result access through IDEA Statica API

### DISCUSSION THREAD

**Discussion by GonzaloNavarroMunoz** [idea-statica/ideastatica-public#243] [Help Q&A] [answered]
Hello! 

Long story short: I have had to review the results of “many” IDEA Statica models and I have not found a “automated” way to do it. So I am opening this issue to suggests the expansion of the current API to be able to access more of the information that is presented in the results of an IDEA Statica connection (welds, bolts, plates…), ideally all of them. 

Right now it is difficult to say why I would want so much information, but I think this is the kind of situation where having the capacity of getting the information will open the door to new ways of working with IDEA Statica that now I cannot see as I do not have that capacity. However it is looks clear that querying some of info will rarely make sense, such as stresses in the plates, over the graphical output already included in the IDEA Statica report. 

But just to start with something “simple”, I would be interested in retrieving all the results for the bolts that are presented in the  table results:
![image](https://user-images.githubusercontent.com/50489727/160877133-297ec77c-eb93-439c-b11b-0fc46263155f.png)

I did the following investigation in that regards: The ConnectionResultData class (from IdeaRS.OpenModel.Connection Namespace) seem to have the results stored for all connections in a project, I understand this is achieved through the List Property of “ConnectionCheckRes” class objects ([https://idea-statica.github.io/iom/iom-api/latest/html/T_IdeaRS_OpenModel_Connection_ConnectionResultsData.htm](https://protect-eu.mimecast.com/s/ts6NCo25KTKGnnxTradWc?domain=idea-statica.github.io) ). I went in to the class page ([https://idea-statica.github.io/iom/iom-api/latest/html/T_IdeaRS_OpenModel_Connection_ConnectionCheckRes.htm](https://protect-eu.mimecast.com/s/7_vzCpY3KUxL22XUpn6pI?domain=idea-statica.github.io) ) and looks like if, for example, I wanted to check results of the bolts I should find them in the List Property of “CheckResBolt” class objects ([https://idea-statica.github.io/iom/iom-api/latest/html/P_IdeaRS_OpenModel_Connection_ConnectionCheckRes_CheckResBolt.htm](https://protect-eu.mimecast.com/s/MqFwCqx35h7rQQmh1gJE0?domain=idea-statica.github.io) ). 
 
However in the page of the class ([https://idea-statica.github.io/iom/iom-api/latest/html/T_IdeaRS_OpenModel_Connection_CheckResBolt.htm](https://protect-eu.mimecast.com/s/2pSpCrk35iDjqqLcDRast?domain=idea-statica.github.io) ), I do not see which properties can store the detailed results that can be viewed in the IDEA Statica model (see markup in image above) and that I would like to access, like the shear and tensile force in each bolt for each Load Case. It looks like the class can store the worst check ("Unity check" property)  for the worst load case, for each bolt that could be identified by the string Property “Name”, not any more of the info mentioned above.

So in summary, it looks like currently only the governing design result (just for the critical LC) can be obtained for each bolt. As I said above, as a start I think it would be quite helpful to retrieve the rest of the table results for each bolt for each LC. 

Thanks very much.

Kind regards,

**Comment by mpospisil** [comment, marked as answer]
Hi Gonzalo,

detailed results are available in our API - see our example in Python

https://github.com/idea-statica/ideastatica-public/tree/main/src/Examples/ConnHiddenCalc/python-examples

https://github.com/idea-statica/ideastatica-public/blob/main/src/Examples/ConnHiddenCalc/python-examples/parameters-anchors.py

you can call 

_# get the results of the check of the first connection
checkResults_json_string = ideaConnectionClient.GetCheckResultsJSON(firstCon.Identifier)_

to get detailed results in JSON format
